diff --git a/node_modules/react-native-navigation/lib/ios/RNNLayoutManager.m b/node_modules/react-native-navigation/lib/ios/RNNLayoutManager.m
index a6c08de..375be0f 100644
--- a/node_modules/react-native-navigation/lib/ios/RNNLayoutManager.m
+++ b/node_modules/react-native-navigation/lib/ios/RNNLayoutManager.m
@@ -2,10 +2,16 @@
 #import "RNNLayoutManager.h"
 #import "RNNLayoutProtocol.h"
 #import "UIViewController+LayoutProtocol.h"
+#import <ReactNativeNavigation/ReactNativeNavigation.h>
 
 @implementation RNNLayoutManager
 
 + (UIViewController *)findComponentForId:(NSString *)componentId {
+	UIViewController* result = [self findChildComponentForParent:[ReactNativeNavigation getRootViewController] ForId:componentId];
+	if (result) {
+		return result;
+	}
+
 	for (UIWindow* window in UIApplication.sharedApplication.windows) {
 		UIViewController* result = [self findChildComponentForParent:window.rootViewController ForId:componentId];
 		if (result) {
diff --git a/node_modules/react-native-navigation/lib/ios/ReactNativeNavigation.h b/node_modules/react-native-navigation/lib/ios/ReactNativeNavigation.h
index 13db8eb..9848cf2 100644
--- a/node_modules/react-native-navigation/lib/ios/ReactNativeNavigation.h
+++ b/node_modules/react-native-navigation/lib/ios/ReactNativeNavigation.h
@@ -17,6 +17,8 @@ typedef UIViewController * (^RNNExternalViewCreator)(NSDictionary* props, RCTBri
 
 + (UIViewController *)findViewController:(NSString *)componentId;
 
++ (UIViewController *)getRootViewController;
+
 + (void)setJSCodeLocation:(NSURL *)jsCodeLocation;
 
 + (RCTBridge *)getBridge;
diff --git a/node_modules/react-native-navigation/lib/ios/ReactNativeNavigation.m b/node_modules/react-native-navigation/lib/ios/ReactNativeNavigation.m
index 2f2ce72..b4655f5 100644
--- a/node_modules/react-native-navigation/lib/ios/ReactNativeNavigation.m
+++ b/node_modules/react-native-navigation/lib/ios/ReactNativeNavigation.m
@@ -12,6 +12,8 @@ @interface ReactNativeNavigation()
 
 @end
 
+UIWindow* rnnMainWindow;
+
 @implementation ReactNativeNavigation
 
 # pragma mark - public API
@@ -40,6 +42,10 @@ + (UIViewController *)findViewController:(NSString *)componentId {
     return [RNNLayoutManager findComponentForId:componentId];
 }
 
++ (UIViewController *)getRootViewController {
+	return rnnMainWindow.rootViewController;
+}
+
 + (void)setJSCodeLocation:(NSURL *)jsCodeLocation {
 	[[ReactNativeNavigation sharedInstance].bridgeManager setJSCodeLocation:jsCodeLocation];
 }
@@ -68,6 +74,7 @@ - (void)bootstrapWithDelegate:(id<RCTBridgeDelegate>)bridgeDelegate launchOption
 
 - (void)bootstrap:(NSURL *)jsCodeLocation launchOptions:(NSDictionary *)launchOptions bridgeManagerDelegate:(id<RCTBridgeDelegate>)delegate {
 	UIWindow* mainWindow = [self initializeKeyWindow];
+	rnnMainWindow = mainWindow;
 	
 	self.bridgeManager = [[RNNBridgeManager alloc] initWithJsCodeLocation:jsCodeLocation launchOptions:launchOptions bridgeManagerDelegate:delegate mainWindow:mainWindow];
 	[RNNSplashScreen showOnWindow:mainWindow];
