diff --git a/node_modules/react-native-interactable/lib/src/InteractableView.js b/node_modules/react-native-interactable/lib/src/InteractableView.js
index acad28a..0f87e0f 100644
--- a/node_modules/react-native-interactable/lib/src/InteractableView.js
+++ b/node_modules/react-native-interactable/lib/src/InteractableView.js
@@ -40,7 +40,7 @@ class WrappedAnimatedInteractableView extends Component {
     }
   }
 
-  componentWillMount() {
+  UNSAFE_componentWillMount() {
     // this.chokeTheBridge();
   }
 
