diff --git a/node_modules/react-native-billing/android/build.gradle b/node_modules/react-native-billing/android/build.gradle
index 4678b6c..95f3817 100644
--- a/node_modules/react-native-billing/android/build.gradle
+++ b/node_modules/react-native-billing/android/build.gradle
@@ -17,6 +17,6 @@ android {
 }
 
 dependencies {
-    compile "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
-    compile "com.anjlab.android.iab.v3:library:${safeExtGet('anjlabIABVersion', '1.0.44')}"
+    implementation "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
+    implementation "com.anjlab.android.iab.v3:library:${safeExtGet('anjlabIABVersion', '1.0.44')}"
 }
